collect_app/src/test/java/org/odk/collect/android/formentry/QuitFormDialogFragmentTest.java

```diff
@@ -0,0 +1,92 @@
+package org.odk.collect.android.formentry;
+
+import android.content.DialogInterface;
+
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.FragmentActivity;
+import androidx.fragment.app.FragmentManager;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.R;
+import org.odk.collect.android.formentry.saving.FormSaveViewModel;
+import org.odk.collect.android.support.RobolectricHelpers;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.shadows.ShadowDialog;
+
+import static android.view.View.GONE;
+import static android.view.View.VISIBLE;
+import static junit.framework.TestCase.assertTrue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.odk.collect.android.support.RobolectricHelpers.mockViewModelProvider;
+import static org.robolectric.Shadows.shadowOf;
+
+@RunWith(RobolectricTestRunner.class)
+public class QuitFormDialogFragmentTest {
+
+    private FragmentActivity activity;
+    private FragmentManager fragmentManager;
+    private QuitFormDialogFragment dialogFragment;
+
+    private FormSaveViewModel viewModel;
+
+    @Before
+    public void setup() {
+        activity = RobolectricHelpers.createThemedActivity(FragmentActivity.class);
+        fragmentManager = activity.getSupportFragmentManager();
+        dialogFragment = new QuitFormDialogFragment();
+
+        viewModel = mockViewModelProvider(activity, FormSaveViewModel.class).get(FormSaveViewModel.class);
+    }
+
+    @Test
+    public void shouldShowCorrectView() {
+        dialogFragment.show(fragmentManager, "TAG");
+        AlertDialog dialog = (AlertDialog) dialogFragment.getDialog();
+
+        assertNotNull(dialog);
+        assertThat(dialog.getButton(DialogInterface.BUTTON_NEGATIVE).getVisibility(), equalTo(GONE));
+        assertThat(dialog.getButton(DialogInterface.BUTTON_POSITIVE).getVisibility(), equalTo(VISIBLE));
+        assertThat(dialog.getButton(DialogInterface.BUTTON_POSITIVE).getText(),
+                equalTo(activity.getString(R.string.do_not_exit)));
+
+        String title = dialogFragment.getTitle(viewModel);
+        dialogFragment.setTitle();
+        assertThat(shadowOf(dialogFragment.getDialog()).getTitle(), equalTo(title));
+    }
+
+    @Test
+    public void dialogIsCancellable() {
+        dialogFragment.show(fragmentManager, "tag");
+        assertThat(shadowOf(dialogFragment.getDialog()).isCancelable(), equalTo(true));
+    }
+
+    @Test
+    public void clickingCancel_shouldDismissTheDialog() {
```

Comment:
Do you still feel we need this test? Happy to talk it over some more!

Reply:
It is something related to the dialog interface. We have, here the positive button set to be the cancel button. So, to check whether the correct button is set as the cancel button and clicking it does as it says (to dismiss the dialog).

Reply:
Ah of course sorry! My mistake. Good test 👍. Let's maybe make it the `NEGATIVE_BUTTON` instead as that's usually the "Cancel" button.

Reply:
Yes, that seems more accurate.